marketplaceItems/appreciate/appResources/appData/appreciate_app.js

```diff
@@ -0,0 +1,947 @@
+                intensityEntityColorMin.green = intensityEntityColorMax.green * 0.4;
+                intensityEntityColorMin.blue = intensityEntityColorMax.blue * 0.4;
+
+                var color = linearScaleColor(currentIntensity, intensityEntityColorMin, intensityEntityColorMax);
+
+                if (intensityMaterialEntity) {
+                    Entities.editEntity(intensityMaterialEntity, {
+                        materialData: JSON.stringify({"materialVersion": 1,
+                            "materials": [
+                                {
+                                    "roughness": 0.0,
+                                    "albedo": [color.red/255, color.green/255, color.blue/255],
+                                }
+                            ]
+                        })
+                    });
+                }
+
+                Entities.editEntity(intensityEntity, {
+                    position: halfwayBetweenHands(),
+                    dimensions: Vec3.multiply(INTENSITY_ENTITY_MAX_DIMENSIONS, currentIntensity),
+                    angularVelocity: Vec3.multiply(currentInitialAngularVelocity,
+                        currentIntensity * ANGVEL_ENTITY_MULTIPLY_FACTOR),
+                    color: color
+                });
+            } else {
+                var props = INTENSITY_ENTITY_PROPERTIES;
+                props.position = halfwayBetweenHands();
+
+                currentInitialAngularVelocity.x = Math.random() *
+                    (INTENSITY_ENTITY_MAX_ANGULAR_VELOCITY.x -
+                    INTENSITY_ENTITY_MIN_ANGULAR_VELOCITY.x) + INTENSITY_ENTITY_MIN_ANGULAR_VELOCITY.x;
+                currentInitialAngularVelocity.y = Math.random() *
+                    (INTENSITY_ENTITY_MAX_ANGULAR_VELOCITY.y -
+                    INTENSITY_ENTITY_MIN_ANGULAR_VELOCITY.y) + INTENSITY_ENTITY_MIN_ANGULAR_VELOCITY.y;
+                currentInitialAngularVelocity.z = Math.random() *
+                    (INTENSITY_ENTITY_MAX_ANGULAR_VELOCITY.z -
+                    INTENSITY_ENTITY_MIN_ANGULAR_VELOCITY.z) + INTENSITY_ENTITY_MIN_ANGULAR_VELOCITY.z;
+                props.angularVelocity = currentInitialAngularVelocity;
+
+                intensityEntity = Entities.addEntity(props, "avatar");
+                
+                var materialProps = INTENSITY_MATERIAL_ENTITY_PROPS;
+                materialProps.parentID = intensityEntity;
+                intensityMaterialEntity = Entities.addEntity(materialProps, "avatar");
+            }
+        } else {
+            if (intensityEntity) {
+                Entities.deleteEntity(intensityEntity);
+                intensityEntity = false;
+            }
+            if (intensityMaterialEntity) {
+                Entities.deleteEntity(intensityMaterialEntity);
+                intensityMaterialEntity = false;
+            }
+            
+            maybeClearUpdateIntensityEntityInterval();
```

Comment:
i like this function.  I've had some issues in the past if I get an uncaught exception during a loop and the interval keeps going until I reset a domain(for a server script at least).  setTimeout generally worked better for me when it was critical, but I don't think you'll have too much trouble.   Just keep an eye out.

Reply:
thank you :)